docs/components/table-data.md

```diff
@@ -0,0 +1,479 @@
+# Table Data
+
+Component that takes care of fetching and rendering data in a table format.
+
+:::tip NOTE
+If you are looking for a simpler table component that does not integrate data fetching, check out [KTableView](/components/table-view).
+:::
+
+<KTableData
+  :fetcher="basicFetcher"
+  :headers="basicHeaders()"
+/>
+
+```html
+<KTableData
+  :fetcher="fetcher"
+  :headers="headers"
+/>
+```
+
+## Underlying KTableView Component
+
+KTableData uses KTableView component providing data fetching layer on top. KTableData supports all KTableView [props](/components/table-view#props) (with the exception of `data` prop, which is replaced with [`fetcher` prop](#fetcher) in KTableData), [slots](/components/table-view#slots) and [events](/components/table-view#events) (with the exception of [pagination events](/components/table-view#pagination-events) which are not emitted by KTableData). `headers` and `hidePaginationWhenOptional` props as well as `update:table-preferences` event are handled slightly differently in KTableData. See below sections for details.
```

Comment:
```suggestion
## Base KTableView Component

KTableData wraps the KTableView component, adding data fetching functionality. 

KTableData supports all [KTableView props](/components/table-view#props) (with the exception of `data` prop, which is replaced with [`fetcher` prop](#fetcher) in KTableData), [slots](/components/table-view#slots) and [events](/components/table-view#events) (with the exception of [pagination events](/components/table-view#pagination-events) which are not emitted by KTableData). `headers` and `hidePaginationWhenOptional` props as well as `update:table-preferences` event are handled slightly differently in KTableData. See below sections for details.
```

Suggesting an update to this paragraph, but you need to break up the sentences still a bit more. That's a long sentence 😅